README.md

```diff
@@ -25,67 +33,80 @@ Configuration is done via a separate file that should be passed to the script on
+## localpool/localdataset:remotepool/remotedataset
+## can include multiple strings separated by a "space"
+## PUSH will push the local to the remote
+## PULL will pull the local to the remote
+REPLICATE_SETS="localpool/localdataset:remotepool/remotedataset"
+
+## option to recursively snapshot children of all datasets listed above
 ## 0 - disable (default)
-## 1 - enable (do so at your own risk)
-ALLOW_ROOT_DATASETS=0
-
-## option to recurrsively snapshot children of
-## all datasets listed above
-## 0 - disable (previous behavior)
 ## 1 - enable
 RECURSE_CHILDREN=0
 
+## if ANY replication task results in an error because of either
+##  - no common snapshot
+##  - snapshots detected on destination
+##  - failure to replicate incremental snapshot
+## this will start the replication from scratch if set to 1
+## and overwrite the existing data on the destination
+## 0 - disable (default)
+## 1 - enable (use at your own risk)
+ALLOW_REPLICATE_FROM_SCRATCH=0
+
+## Allow replication of root datasets
+## if you specify root datasets above and do not toggle this setting the
+## script will generate a warning and skip replicating root datasets
+## 0 - disable (default)
+## 1 - enable (use at your own risk)
+ALLOW_ROOT_DATASETS=0
+
 ## number of snapshots to keep of each dataset
-## snaps in excess of this number will be expired
-## oldest deleted first...must be 2 or greater
+## older snapshots will be deleted
 SNAP_KEEP=2
 
-## number of logs to keep in path ... logs will be
-## deleted in order of age with oldest going first
-LOG_KEEP=10
+## number of logs to keep
+## older logs will be deleted
+LOG_KEEP=5
 
-## where you want your log files
-## and gnu tar incremental snaphots
-LOGBASE=/root/logs
-
-## ip address or hostname of a remote server
-## this variable may be referenced in the
-## additional settings below
-##
-## this should not be used for local replication
-## and could be commented out and ignored
-REMOTE_SERVER='192.168.100.2'
+## log files directory (defaults to script path)
+SCRIPT=$(readlink -f "$0")
+SCRIPTPATH=$(dirname "${SCRIPT}")
+LOGBASE="${SCRIPTPATH}/logs"
 
 ## command to check health of remote host
 ## a return code of 0 will be considered OK
-##
-## this is not used for local replication
-## and could be commented out and ignored
+## comment out for local only replication
 REMOTE_CHECK="ping -c1 -q -W2 ${REMOTE_SERVER}"
 
-## pipe to your remote host...the pool/snap
-## DO NOT INCLUDE THE PIPE (|) CHARACTER
-## fs names from this host will be used on the remote
-##
-## for increased transfer speed you may want to specifically
-## enumerate your prefered cipher order in your ssh command:
-## ssh -c arcfour256,arcfour128,blowfish-cbc,aes128-ctr,aes192-ctr,aes256-ctr
-##
-## for local replication do not
-## call ssh or reference a remote server
-RECEIVE_PIPE="ssh ${REMOTE_SERVER} zfs receive -vFd"
-
-## path to zfs binary
-ZFS=/sbin/zfs
+## path to zfs binary (only command for now)
+ZFS=zfs
```

Comment:
Should this be bare without a path?

Reply:
This is fine. I just used the command because the original file path did not work. Is this where zfs is usually found?

Reply:
Probably /usr/local/sbin/zfs for most systems. But if differs on some so I left it as simply zfs

Reply:
On hosts where it's part of the base it will be in `/sbin/zfs`, but that's a fair point. If it's not found by the $PATH folks can fix it :)